### PR TITLE
object-cache.php drop-in copy now able to be aborted

### PR DESCRIPTION
**Pull request by polevaultweb**
Resolves https://github.com/deliciousbrains/spinupwp-plugin/issues/53

### CODE REVIEW

File: src/Plugin.php [hunk 1]
```diff
@@ -67,15 +69,47 @@ public static function install() {
...
+		return $result;
+	}
+
+	/**
+	 * @param bool $check
+	 *
+	 * @return bool
+	 */
+	public function should_use_object_cache_dropin( $check ) {
+		if ( defined( 'RedisCachePro\Version' ) ) {
```
[the remark this answers is not in the document]
Comment (A5hleyRich) [reply src/Plugin.php:107]: I don't think so, as our version of object-cache.php sets a few preferred defaults on Spinup.

File: src/Plugin.php [hunk 2]
```diff
@@ -67,15 +69,47 @@ public static function install() {
...
+	public static function update_object_cache_dropin() {
+		if ( ! apply_filters( 'spinupwp_should_use_object_cache_dropin', true ) ) {
+			return false;
+		}
+
+		$wpcontent_dir = untrailingslashit( WP_CONTENT_DIR );
+		$plugin_path   = untrailingslashit( dirname( dirname( __DIR__ ) ) );
+
 		if ( file_exists( $wpcontent_dir . '/object-cache.php' ) ) {
 			@unlink( $wpcontent_dir . '/object-cache.php' );
```
[the remark this answers is not in the document]
Comment (A5hleyRich) [reply src/Plugin.php:89]: 👍

File: src/Plugin.php [hunk 3]
```diff
@@ -67,15 +69,47 @@ public static function install() {
 			@copy( $plugin_path . '/mu-plugins/spinupwp-debug-log-path.php', $wpmu_dir . '/spinupwp-debug-log-path.php' );
 		}
 
+		self::update_object_cache_dropin();
```
Comment (A5hleyRich) [inline src/Plugin.php:72]: `$wpcontent_dir` is no longer used.

**Review by A5hleyRich — CHANGES_REQUESTED** [review]
@polevaultweb does the alignment look off to you? It's all kind of weird on the GH diff.

**Comment by polevaultweb** [comment]
@A5hleyRich back to you

**Comment by A5hleyRich** [comment]
💯

**Pull request merged**